network_client.go

```diff
@@ -51,9 +52,21 @@ func (nc *NetworkClient) send(topic string, partition int32, batch []*ProducerRe
 	}
 	responseChan := nc.selector.Send(leader, request)
 
-	if nc.requiredAcks > 0 {
-		go listenForResponse(topic, partition, batch, responseChan)
-	} else {
+	go nc.listenForResponse(topic, partition, batch, responseChan)
+}
+
+func (nc *NetworkClient) listenForResponse(topic string, partition int32, batch []*ProducerRecord, responseChan <-chan *rawResponseAndError) {
+	response := <-responseChan
```

Comment:
With required acks == 0 Kafka broker won't send any response, thus this line will block if required acks == 0, no? I think you should add that "if" statement above back.